### PR TITLE
feat(forms): add tokens and update shim

### PR DESCRIPTION
**Pull request by dtsanevmw**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: CDE-668

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: projects/angular/src/forms/styles/_properties.forms.scss [hunk 1]
```diff
@@ -12,29 +12,38 @@
...
       --clr-forms-text-color: var(--clr-color-neutral-1000);
       --clr-forms-invalid-color: var(--clr-color-danger-800);
+      --clr-forms-invalid-text-color: var(--clr-color-danger-800);
       --clr-forms-valid-color: var(--clr-color-success-700);
       --clr-forms-valid-text-color: var(--clr-color-success-900);
       --clr-forms-subtext-color: var(--clr-color-neutral-600);
       --clr-forms-placeholder-color: var(--clr-global-color-construction-600);
       --clr-forms-border-color: #{forms-variables.$clr-forms-border-color};
-      --clr-forms-focused-color: var(--clr-color-action-600);
+      --clr-forms-focused-color: var(--clr-color-action-500);
```
Comment (valentin-mladenov) [inline projects/angular/src/forms/styles/_properties.forms.scss:28]: Why is that change needed. It will affect only clarity theme.

File: projects/ui/src/shim.cds-core.scss [hunk 2]
```diff
@@ -441,23 +441,36 @@
   /* Forms */
 
   // Structure
+  // --clr-forms-baseline: var(--cds-global-space-5);
   --clr-form-disabled-background-color: var(--cds-alias-status-disabled-tint); // var(--clr-color-neutral-400);
 
   // Generic
+  --clr-forms-disabled-color: var(--cds-alias-status-disabled);
```
Comment (valentin-mladenov) [inline projects/ui/src/shim.cds-core.scss:448]:
```suggestion
  --clr-forms-disabled-color: var(--cds-alias-typography-disabled, var(--cds-alias-status-disabled));
```

I believe this variable must be mapped to typography disabled.

**Review by valentin-mladenov — CHANGES_REQUESTED** [review]
Check my comments.

File: projects/angular/src/forms/styles/_properties.forms.scss (hunk 1 above)
Comment (dtsanevmw) [reply projects/angular/src/forms/styles/_properties.forms.scss:28]: Yeah it was not needed, I was just looking at something when I changed it

File: projects/ui/src/shim.cds-core.scss (hunk 2 above)
Comment (dtsanevmw) [reply projects/ui/src/shim.cds-core.scss:448]: We have both variables, one is used for text and the other for other things like color of icons borders etc..

File: projects/angular/src/forms/styles/_containers.clarity.scss [hunk 3]
```diff
@@ -136,7 +151,11 @@
   }
 
   .clr-validate-icon {
-    @include mixins.min-equilateral(forms-variables.$clr-forms-icon-size);
+    @include mixins.equilateral-var(
```
Comment (valentin-mladenov) [inline projects/angular/src/forms/styles/_containers.clarity.scss:154]: Different equilateral mixin should be used at this place, probably `min-equilateral-var` should be created to handle the changes

File: projects/ui/src/shim.cds-core.scss [hunk 4]
```diff
@@ -441,23 +441,34 @@
...
+  --clr-forms-focused-color: var(--cds-alias-object-interaction-background-highlight); // var(--clr-color-action-500);
+  --clr-forms-icon-size: var(--cds-global-space-9);
+  --clr-forms-subtext-margin-top: var(--cds-global-space-3);
+
+  --clr-forms-subtext-disabled-color: var(--cds-alias-typography-disabled); // #{$clr-forms-subtext-disabled-color};
+  --clr-forms-border-disabled-color: var(--cds-alias-object-interaction-color-disabled); // #{$clr-forms-border-disabled-color};
+  --clr-forms-text-disabled-color: var(--cds-alias-typography-disabled); // #{$clr-forms-text-disabled-color};
+  --clr-forms-label-disabled-color: var(--cds-alias-typography-disabled); // #{$clr-forms-label-disabled-color};
+  --clr-forms-icon-disabled-color: var(--cds-alias-status-disabled);
+  --clr-forms-placeholder-disabled-color: var(--cds-alias-typography-disabled);
```
Comment (valentin-mladenov) [inline projects/ui/src/shim.cds-core.scss:467]: Add a fallback for the missing `var(--cds-alias-typography-disabled)`.

```suggestion
  --clr-forms-subtext-disabled-color: var(--cds-alias-typography-disabled, var(--cds-alias-status-disabled)); // #{$clr-forms-subtext-disabled-color};
  --clr-forms-border-disabled-color: var(--cds-alias-object-interaction-color-disabled); // #{$clr-forms-border-disabled-color};
  --clr-forms-text-disabled-color: var(--cds-alias-typography-disabled, var(--cds-alias-status-disabled)); // #{$clr-forms-text-disabled-color};
  --clr-forms-label-disabled-color: var(--cds-alias-typography-disabled, var(--cds-alias-status-disabled)); // #{$clr-forms-label-disabled-color};
  --clr-forms-icon-disabled-color: var(--cds-alias-status-disabled);
  --clr-forms-placeholder-disabled-color: var(--cds-alias-typography-disabled, var(--cds-alias-status-disabled));
```

File: projects/ui/src/shim.cds-core.scss [hunk 5]
```diff
@@ -441,23 +441,34 @@
   /* Forms */
 
   // Structure
+  // --clr-forms-baseline: var(--cds-global-space-5);
```
Comment (valentin-mladenov) [inline projects/ui/src/shim.cds-core.scss:444]: is this comment useful somehow?

**Review by valentin-mladenov — CHANGES_REQUESTED** [review]
Check my comments.

File: projects/ui/src/shim.cds-core.scss (hunk 4 above)
Comment (dtsanevmw) [reply projects/ui/src/shim.cds-core.scss]: It should point to `var(--cds-global-color-construction-400)` as `--cds-alias-status-disabled` is different color but yeah I forgot these fallbacks. Thanks :)

File: projects/ui/src/shim.cds-core.scss (hunk 5 above)
Comment (dtsanevmw) [reply projects/ui/src/shim.cds-core.scss]: Missed to remove it. Thanks!

File: projects/ui/src/shim.cds-core.scss (hunk 4 above)
Comment (valentin-mladenov) [reply projects/ui/src/shim.cds-core.scss]: Probably `var(--cds-global-typography-color-400)` will be better for the text variables.

Comment (valentin-mladenov) [reply projects/ui/src/shim.cds-core.scss]: The text part should fall to some `var(--cds-global-typography-color-XXX)`, but I don't know do we have those.

**Review by valentin-mladenov — APPROVED** [review]
LGTM

File: projects/angular/src/forms/styles/_properties.forms.scss [hunk 6]
```diff
@@ -12,12 +12,14 @@
...
     :root {
       // Structure
+      --clr-forms-baseline: #{variables.$clr_baselineRem_0_25};
       --clr-form-disabled-background-color: var(--clr-color-neutral-400);
 
       // Generic
       --clr-forms-label-color: var(--clr-color-neutral-800);
       --clr-forms-text-color: var(--clr-color-neutral-1000);
       --clr-forms-invalid-color: var(--clr-color-danger-800);
+      --clr-forms-invalid-text-color: var(--clr-color-danger-800);
```
Comment (kevinbuhmann) [inline projects/angular/src/forms/styles/_properties.forms.scss:22]: I think we should file a ticket to do this rename in a breaking change PR for v16 and just keep using the exiting CSS custom property instead of duplicating it. The code in this PR uses both of these and it could cause confusion.

Comment (dtsanevmw) [reply projects/angular/src/forms/styles/_properties.forms.scss:22]: It's equal for the default theme but different when using core. We can skip adding these to _properties file and it would use the custom property if available or the default value for example?

Comment (Jinnie) [reply projects/angular/src/forms/styles/_properties.forms.scss:22]: I believe we can go on with this and do a cleanup after the token migration.
We can add and //TODO: ATM comment to keep track of this intent.

**Review by Jinnie — COMMENTED** [review]
We have unexpected changes on two NG screenshots: datalist (placeholder) and datepicker (some borders)

**Review by Jinnie — APPROVED** [review]
It's a foundational PR that may be adjusted while dependent components are being merged.
Thanks you!
